### PR TITLE
Fix memory leak, improve logging.

### PR DESCRIPTION
**Pull request by olonho**

### CODE REVIEW

File: runtime/src/main/cpp/Memory.cpp [hunk 1]
```diff
@@ -639,11 +640,47 @@ inline void initThreshold(MemoryState* state, uint32_t gcThreshold) {
...
+  if (relativeName) konan::free(relativeName);
+}
+
+void dumpContainerContent(ContainerHeader* container) {
+  if (isAggregatingFrozenContainer(container)) {
+    MEMORY_LOG("%s aggregating container %p with %d objects rc=%d\n",
+               colorNames[container->color()], container, container->objectCount(), container->refCount());
+    ContainerHeader** subContainer = reinterpret_cast<ContainerHeader**>(container + 1);
+    for (int i = 0; i < container->objectCount(); ++i) {
+      ObjHeader* obj = reinterpret_cast<ObjHeader*>(subContainer + 1);
```
Comment (SvyatoslavScherbina) [inline runtime/src/main/cpp/Memory.cpp]: It this correct?

File: runtime/src/main/cpp/Memory.cpp [hunk 2]
```diff
@@ -639,11 +640,47 @@ inline void initThreshold(MemoryState* state, uint32_t gcThreshold) {
...
+      MEMORY_LOG("    object %p of type %p: ", obj, obj->type_info());
+      dumpObject(obj, 4);
+    }
+  } else {
+    MEMORY_LOG("%s regular %s%scontainer %p with %d objects rc=%d\n",
```
Comment (SvyatoslavScherbina) [inline runtime/src/main/cpp/Memory.cpp]: Missing spaces.

File: runtime/src/main/cpp/Memory.cpp (hunk 1 above)
Comment (olonho) [reply runtime/src/main/cpp/Memory.cpp]: Why not, supercontainer contains references to frozen objects containers.

File: runtime/src/main/cpp/Memory.cpp (hunk 2 above)
Comment (olonho) [reply runtime/src/main/cpp/Memory.cpp]: Not really, we add spaces to strings.

File: runtime/src/main/cpp/Memory.cpp (hunk 1 above)
Comment (SvyatoslavScherbina) [reply runtime/src/main/cpp/Memory.cpp]: `obj` seems to be the same on all loop iterations, doesn't it?
Is it really located at `container + sizeof(ContainerHeader) + sizeof(ContainerHeader*)`?

**Pull request merged**